src/ScriptProfile.cc

```diff
@@ -47,9 +47,14 @@ void ScriptProfile::Report(FILE* f) const
 
 	std::string ftype = is_BiF ? "BiF" : func->GetType()->FlavorString();
 
-	fprintf(f, "%s\t%s\t%s\t%d\t%.06f\t%.06f\t%" PRIu64 "\t%" PRIu64 "\n", Name().c_str(),
+	std::string call_stacks;
+	for ( const auto& [s, count] : Stacks() )
```

Comment:
Only thought is whether `Stacks()` should be `RenderedCallStack()` and the string formatting a bit hidden.

The other one: This is producing pretty long lines when testing real workload? Does it make the existing output which might have been meant to be consumable as a table on the terminal difficult to use?

Reply:
Yeah, I wonder about that too.  I also chatted with Tim about how conceptually it seems to me there should be three additional columns, one for number-of-calls-per-tree (which is what this branch currently implements, and what Flame Graphs use), one for CPU-per-tree, and one for memory-per-tree.  So in my ideal world, we'd (1) have all three of those, but also (2) have a different/additional flag for turning them on, so it's still easy to get the simpler works-well-in-a-terminal-window version that's the current behavior.

Reply:
If that's the desire, then I'd propose 4 columns. One with the call stacks, and then one for each of the calls, cpu time, and memory. It'd add a little bit of time to post-processing, but it would reduce the amount of duplicate data in the output.

Reply:
The latest commit added a `--profile-scripts-traces` argument to turn the trace output on and off. It also splits out the call counts and adds CPU/memory data. The new output looks like this from `$TRACES/wikipedia.trace`:

```
#fields function        location        type    ncall   tot_CPU child_CPU       tot_Mem child_Mem       stacks        stack_calls     stack_CPU       stack_memory
#types  string  string  string  count   interval        interval        count   count   string  string  string        string
Log::write      /Users/tim/Desktop/projects/zeek/scripts/base/frameworks/logging/./main.zeek:839        function      83      0.001765        0.001309        163840  163840          Log::write;zeek_init;PacketFilter::install|Log::write;http_message_done|Log::write;dns_end|Log::write;connection_state_remove;DNS::finalize_dns;DNS::log_unmatched_msgs;DNS::log_unmatched_msgs_queue|Log::write;connection_state_remove;DNS::finalize_dns|Log::write;connection_state_remove     1|14|14|12|8|34 0.000152|0.000257|0.000260|0.000293|0.000128|0.000675   65536|16384|16384|0|0|65536
```